pkg/kubelet/kubelet_node_status.go

```diff
@@ -341,23 +341,32 @@ func (kl *Kubelet) setNodeAddress(node *api.Node) error {
 }
 
 func (kl *Kubelet) setNodeStatusMachineInfo(node *api.Node) {
+	// Note: avoid blindly overwriting the capacity in case opaque
+	//       resources are being advertised.
+	if node.Status.Capacity == nil {
+		node.Status.Capacity = api.ResourceList{}
```

Comment:
@ConnorDoyle -- apologies for the delay, too much going on.

During kubelet admission, we call a registered set of admit handlers to do admission checks that are registered.

see: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L724

this function ends up using `getNodeAnyway`, which falls through to this function you modified eventually, so I think we are ok for the area that would actually be impacted.